### PR TITLE
Try to convert all string columns to int/floats in design matrix

### LINKED ISSUE #11430: JINJA evaluation have changed for DESIGN_MATRIX
### What happened? (You can include a screenshot if it helps explain)

Using jinja template below creates unexpected output:

Template:
```
PORO
-- Constant porosity of 0.3 throughout all 300 grid cells
    300*{{10 * parameters.FIELD_PROPERTIES.POROSITY}} /

PERMX
-- The layers have perm. 500mD, 50mD and 200mD, respectively.
        100*500 100*{{100 * parameters.DESIGN_MATRIX.X_MID_PERMEABILITY}} 100*200 /
```

results in 

```
PORO
-- Constant porosity of 0.3 throughout all 300 grid cells
    300*3.121481301708695 /

PERMX
-- The layers have perm. 500mD, 50mD and 200mD, respectively.
	100*500 100*0.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.50.5 100*200 /
```

### What did you expect to happen?

Expect that DESIGN_MATRIX and GEN_KW evaluation works the same, and that the math in the jinja part is evaluated. 

In the stable version, it produces, as expected, this:
```
PORO
-- Constant porosity of 0.3 throughout all 300 grid cells
    300*1.0270276627188744 /

PERMX
-- The layers have perm. 500mD, 50mD and 200mD, respectively.
	100*500 100*50.0 100*200 /
```

### Steps to reproduce

```Python
Run the example in https://github.com/equinor/ert-testdata/tree/data-for-open-old-storage/version-6 (flow_example + design matrix) on bleeding/main vs stable.
```

### Environment where bug has been observed

- [x] Python 3.11
- [ ] Python 3.12
- [ ] macOS
- [x] RHEL8
- [ ] RHEL9
- [x] Local queue
- [ ] LSF queue
- [ ] Slurm queue
- [ ] OpenPBS queue

### PR DESCRIPTION
**Pull request by larsevj**
**Issue**
Resolves #11430 

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'just rapid-tests'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Add backport label to latest release (format: 'backport release-branch-name')

### CODE REVIEW

File: tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py [hunk 1]
```diff
@@ -433,3 +433,27 @@ def test_whitespace_is_stripped_from_string_parameters(tmp_path):
...
+
+def test_that_numeric_string_columns_are_converted(tmp_path):
+    realizations = [0, 1, 4]
+    design_path = tmp_path / "design_matrix.xlsx"
+    design_matrix_df = pl.DataFrame(
+        {
+            "REAL": realizations,
+            "a": ["1", "2", "3"],
+            "b": [0, "2.2", "0.1"],
+            "c": ["low", "high", "medium"],
```
Comment (berland) [inline tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py]: Can you add a line (a column) with both strings and number such that that corner-case is documented?

File: tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py [hunk 2]
```diff
@@ -433,3 +433,27 @@ def test_whitespace_is_stripped_from_string_parameters(tmp_path):
...
+        },
```
Comment (berland) [inline tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py:447]: Can `numpy.nan` and/or `None` occur here?

File: src/ert/config/design_matrix.py [hunk 3]
```diff
@@ -403,6 +405,26 @@ def _read_defaultssheet(
         }
 
 
+def convert_numeric_string_columns(df: pl.DataFrame) -> pl.DataFrame:
+    """Automatically convert string columns to numeric (int or float) where possible"""
+    for col in df.columns:
```
Comment (xjules) [inline src/ert/config/design_matrix.py]: this looks a bit better:
`for col, dtype in zip(df.columns, df.dtypes):`

File: src/ert/config/design_matrix.py [hunk 4]
```diff
@@ -403,6 +405,26 @@ def _read_defaultssheet(
...
+        if df[col].dtype == pl.String:
+            try:
+                temp = df[col].cast(pl.Int64, strict=True)
```
Comment (xjules) [inline src/ert/config/design_matrix.py]: this can be inlined:
`df = df.with_columns(pl.col(col).cast(pl.Int64, strict=True).alias(col))`

the same for the bottom one

File: tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py (hunk 2 above)
Comment (larsevj) [reply tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py:447]: Adding for instance `"d": [np.nan, None, "some"],` raises validation error as the two first elements are treated as missing values and disallowed.
If they are string values in the excel file, e.g. `"d": ["np.nan", "None", "some"],`, then only validation error occurs on  the "None" value due to hardcoded check agains "None" and "nan" strings. Using "np.nan" will pass through so it is perfectly valid with, `"d": ["np.nan", "low", "high"],`, but not sure how far to stretch the validation.

Comment (berland) [reply tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py:447]: `"d": ["np.nan", "low", "high"]` should pass as strings. The hardcoded checks are probably inherited from semeio? Keep as is 👍🏻

File: src/ert/config/design_matrix.py (hunk 3 above)
Comment (larsevj) [reply src/ert/config/design_matrix.py]: Done

File: src/ert/config/design_matrix.py (hunk 4 above)
Comment (larsevj) [reply src/ert/config/design_matrix.py]: Done

File: tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py (hunk 1 above)
Comment (larsevj) [reply tests/ert/unit_tests/sensitivity_analysis/test_design_matrix.py]: Done
